src/groupeddataframe/splitapplycombine.jl

```diff
@@ -926,11 +1024,25 @@ isagg(p::Pair) = check_aggregate(last(p)) isa AbstractAggregate && first(p) isa
 
 const MULTI_COLS_TYPE = Union{AbstractDataFrame, NamedTuple, DataFrameRow, AbstractMatrix}
 
+function _agg2idx_map_helper(idx, idx_agg)
+    agg2idx_map = fill(-1, length(idx))
+    aggj = 1
+    @inbounds for (j, idxj) in enumerate(idx)
+        while idx_agg[aggj] != idxj
+            aggj += 1
+            @assert aggj <= length(idx_agg)
```

Comment:
I was thinking about a single check at the end of the loops. This one is going to be quite costly, right?

Reply:
I would not do it. This loop is cheap relative to all other work we do in combine - even for very large tables it is just a single pass in a loop. Note that it is only executed if vectors and single values are mixed (so we do a lot of more expensive operations anyway).

Now what is a consequence of avoiding this check? the `while idx_agg[aggj] != idxj` loop might go to infinity and/or produce a segfault. I think the benefit is not worth it. I would either leave this assertion or drop `@inbounds`.

Reply:
OK, as you like.

Reply:
I have benchmarked it. The difference in speed with `@assert` is not noticeable:
```
julia> x = 1:10^6;

julia> function f1(x, l)
           s = 0.0
           i = 1
           while i < l
               s += sin(i) # do something that will not be optimized out
               @assert i <= length(x)
               i += 1
           end
           s
       end;

julia> function f2(x, l)
           s = 0.0
           i = 1
           while i < l
               s += sin(i) # do something that will not be optimized out
               i += 1
           end
           s
       end;

julia> @benchmark f1($x, 10^6)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     16.509 ms (0.00% GC)
  median time:      16.773 ms (0.00% GC)
  mean time:        16.828 ms (0.00% GC)
  maximum time:     24.072 ms (0.00% GC)
  --------------
  samples:          298
  evals/sample:     1

julia> @benchmark f2($x, 10^6)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     16.465 ms (0.00% GC)
  median time:      16.782 ms (0.00% GC)
  mean time:        16.839 ms (0.00% GC)
  maximum time:     23.997 ms (0.00% GC)
  --------------
  samples:          297
  evals/sample:     1
```